codegen/src/main/java/software/amazon/awssdk/codegen/AddOperations.java

```diff
@@ -179,6 +180,12 @@ public Map<String, OperationModel> constructOperations() {
             if (input != null) {
                 String originalShapeName = input.getShape();
                 String inputShape = namingStrategy.getRequestClassName(operationName);
+
+                if (isQueryProtocolWithExplicitStringPayload(c2jShapes, c2jShapes.get(inputShape))) {
```

Comment:
I don't like the idea of doing validation in here since we're not doing it for anything else. Can we do this validation as a preprocessor instead? https://github.com/aws/aws-sdk-java-v2/blob/ea1bb9b7e725c0a1c53fe34336ae7902d3a8924b/codegen/src/main/java/software/amazon/awssdk/codegen/customization/CodegenCustomizationProcessor.java#L21 similar to https://github.com/aws/aws-sdk-java-v2/blob/ea1bb9b7e725c0a1c53fe34336ae7902d3a8924b/codegen/src/main/java/software/amazon/awssdk/codegen/customization/processors/NewAndLegacyEventStreamProcessor.java#L34

Reply:
makes sense, moving to preproccesor